pylhc/constants/autosix.py

```diff
@@ -0,0 +1,138 @@
+"""
+Constants: Autosix
+----------------------------------
+
+Collections of constants and paths used in autosix.
+
+:module: constants.autosix
+:author: jdilly
+
+"""
+from pathlib import Path
+
+from generic_parser import DotDict
+
+from pylhc.constants.external_paths import SIXDESK_UTILS, MADX_BIN
+
+# Program Paths ----------------------------------------------------------------
+SETENV_SH = SIXDESK_UTILS / "set_env.sh"
+MAD_TO_SIXTRACK_SH = SIXDESK_UTILS / "mad6t.sh"
+RUNSIX_SH = SIXDESK_UTILS / "run_six.sh"
+RUNSTATUS_SH = SIXDESK_UTILS / "run_status"
+DOT_PROFILE = SIXDESK_UTILS / "dot_profile"
+SIXDB = SIXDESK_UTILS.parent / "externals" / "SixDeskDB" / "sixdb"
+SIXDESKLOCKFILE = "sixdesklock"
+
+# Constants and Requirements ---------------------------------------------------
+
+HEADER_BASEDIR = "BASEDIR"
+
+DEFAULTS = dict(
+    python="python3",
+    da_turnstep=100,
+    executable=MADX_BIN,
+)
+
+# Sixenv ---
+SIXENV_REQUIRED = ["TURNS", "AMPMIN", "AMPMAX", "AMPSTEP", "ANGLES"]
+SIXENV_DEFAULT = dict(
+    RESUBMISSION=0,  # 0: never, 1: if fort.10, 2: always
+    PLATFORM="HTCondor",
+    LOGLEVEL=0,  # 0: basic + errors , 1: + info, >=2: + debug
+    FIRSTSEED=1,
+    LASTSEED=60,
+    ENERGY="col",  # 'col' or 'inj'
+    NPAIRS=30,  # 1-32 particle pairs
+    EMITTANCE=3.75,  # normalized emittance
+    DIMENSIONS=6,  # Phase-Space dimensions
+    WRITEBINS=500,
+)
+SEED_KEYS = ["FIRSTSEED", "LASTSEED"]
+
+
+# Stages ---
+STAGE_ORDER = [
+    "create_jobs",
+    "submit_mask",
+    "check_input",
+    "submit_sixtrack",
+    "check_sixtrack_output",
+    "sixdb_load",
+    "sixdb_cmd",
+    "post_process",
+    "final",
+]
+STAGES = DotDict({key: key for key in STAGE_ORDER})
+
+
+# SixDB and Postprocess ---
+
+HEADER_NTOTAL, HEADER_INFO, HEADER_HINT = "NTOTAL", "INFO", "HINT"
+MEAN, STD, MIN, MAX, N = "MEAN", "STD", "MIN", "MAX", "N"
+SEED, ANGLE, ALOST1, ALOST2, AMP = "SEED", "ANGLE", "ALOST1", "ALOST2", "A"
+
+
+# Workspace Paths --------------------------------------------------------------
+
+# Input ---
+def get_workspace_path(jobname: str, basedir: Path) -> Path:
```

Comment:
I don't know how I feel about non-constants in a `constants` module, maybe this should all be in a `utils` in `sixdesk_tools`?

Reply:
yeah, I had this discussion with Lukas before. I'd argue that paths are also constants as the folder names are constants.
similar to `get_error_colum(colum): return f"ERR{column"}` I'd almost argue is a constant. Otherwise I'd have to make the folder names a constant and then move the function somewhere else... which does not increase readability